application/src/main/java/org/togetherjava/tjbot/commands/tags/TagCommand.java

```diff
@@ -46,20 +50,40 @@ public void onSlashCommand(SlashCommandInteractionEvent event) {
         String id = Objects.requireNonNull(event.getOption(ID_OPTION)).getAsString();
         OptionMapping replyToUserOption = event.getOption(REPLY_TO_USER_OPTION);
 
-        if (tagSystem.handleIsUnknownTag(id, event)) {
+        if (tagSystem.handleIsUnknownTag(id, event, super::generateComponentId,
+                replyToUserOption)) {
             return;
         }
 
-        ReplyCallbackAction message = event
-            .replyEmbeds(new EmbedBuilder().setDescription(tagSystem.getTag(id).orElseThrow())
-                .setFooter(event.getUser().getName() + " • used " + event.getCommandString())
-                .setTimestamp(Instant.now())
-                .setColor(TagSystem.AMBIENT_COLOR)
-                .build());
+        sendTagReply(event, event.getUser().getName(), id, Optional.of(event.getCommandString()),
+                Optional.ofNullable(replyToUserOption)
+                    .map(OptionMapping::getAsUser)
+                    .map(IMentionable::getAsMention));
+    }
 
-        if (replyToUserOption != null) {
-            message = message.setContent(replyToUserOption.getAsUser().getAsMention());
+    @Override
+    public void onButtonClick(ButtonInteractionEvent event, List<String> args) {
+        if (!TagSystem.TAG_SUGGESTION_INDICATOR.equals(args.get(0))) {
+            return;
         }
-        message.queue();
+
+        sendTagReply(event, event.getUser().getName(), args.get(1), Optional.empty(),
+                Optional.ofNullable(args.get(2)));
+    }
+
+    /**
+     * Sends the reply for a successfull /tag use (i.e. the given tag exists)
+     */
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
```

Comment:
why suppress? dont do it, use `@Nullable` instead.

Reply:
then I'd have to use a bunch of ternaries inside which would look really ugly

Reply:
done